flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/invokable/operator/co/CoInvokable.java

```diff
@@ -84,7 +84,16 @@ public void invoke() throws Exception {
 				next = recordIterator.next(reuse1, reuse2);
 			} catch (IOException e) {
 				if (isRunning) {
-					throw e;
+					throw new RuntimeException("Could not read next record due to: "
+							+ StringUtils.stringifyException(e));
```

Comment:
I think the `stringifyException` is an antipattern. Exceptions should be chained, the stack trace should not go into the message. I would actually go as far as removing `StringUtils.stringifyException`, because it encourages bad use of exceptions.